gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/PathRetractionStrategyTest.java

```diff
@@ -191,6 +206,9 @@ public void doTest() {
                 {__.V().as("a").optional(bothE().dedup().as("b")).
                         choose(select("b"), select("a","b"), project("a").by(select("a"))),
                         "[[[a, b]], [[a, b]], [[a, b]], [[[a, b]]], [[a, b]]]", null},
+                {__.V().as("a").out().program(labeledPathVertexProgram).select("a"), "[]", null},
+                {__.V().as("a").out().program(pathVertexProgram).select("a"), "[]", null},
+                {__.V().as("a").out().program(emptyRequirementsVertexProgram).select("a"), "[[]]", null}
```

Comment:
Yeah, that is confusing.  It is correctly asserting with that `[]` but it is non-obvious. I replaced those with a constant and then reworked the traversals.